src/components/CardContainer/CardContainer.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import Card from "../Card/Card"
+import UniqueRandomArray from "unique-random-array"
+
+class CardContainer extends React.Component {
+    constructor(props) {
+        super(props)
+
+        this.state = {
+            dogs: []
+        }
+    }
+
+    componentDidMount() {
+        this.getDogs()
+    }
+
+    getDogs = () => {
+        fetch('http://localhost:3000/dogs', {
+            method: 'get'
+        })
+            .then((data) => data.json())
+            .then((res) => {
+                if (res.success) {
+                    return res.data
+                } else {
+                    return res.message
+                }
+            })
+            .then((res) => {
+                    let random = UniqueRandomArray(res)
+                    this.setState({
+                        dogs: [random(), random()]
```

Comment:
I think good practice is to take a copy of the whole component state, change the one aspect of it that we want to change, and call setState().

In this instance it's not much difference at all, but if we had other state that we wanted to persist then this current implementation would wipe out everything else. Happy to show an example of what I mean

Reply:
on it

Reply:
on it